src/Orchard.Environment.Extensions/ExtensionManager.cs

```diff
@@ -82,7 +82,12 @@ public IExtensionInfo GetExtension(string extensionId)
         {
             GetExtensions(); // initialize
 
-            return _extensionsById[extensionId];
+            if (_extensionsById.ContainsKey(extensionId))
+            {
+                return _extensionsById[extensionId];
+            }
+
+            return null;
```

Comment:
Ok. Here also, when the current selected theme is no more present on the disk, it was throwing an exception because the dictionary key doesn't exist.

Ok but didn't find a `NotFoundExtensionInfo` class already defined. So, do you mean i have to create such a class (e.g in `Extensions.Abstractions`)? Which embed e.g a `NotFoundFileInfo` and a `NotFoundManifestInfo`. Right? If so, will do it this night asap.